helix-core/src/main/java/org/apache/helix/controller/stages/task/TaskSchedulingStage.java

```diff
@@ -81,13 +81,16 @@ public void process(ClusterEvent event) throws Exception {
     // Reset current INIT/RUNNING tasks on participants for throttling
     cache.resetActiveTaskCount(currentStateOutput);
 
+    ClusterStatusMonitor clusterStatusMonitor =
+        event.getAttribute(AttributeName.clusterStatusMonitor.name());
```

Comment:
Should we just do a null check here and remove null checks in other places (e.g. JobMonitor)?

Reply:
Since @junkaixue also commented about doing the work in `ClusterStatusMonitor` instead, we should definitely do the null check here. Thanks for the suggestion.